File: skill-factor-ui/src/api/factor.ts
import type { Factor } from '../types/factor';
import type { Scenario } from '../types/scenario';
import { USE_MOCK, apiFetch, mockDelay } from './client';
import { mockFactors } from '../mock/factors';

export async function generateFactors(scenario: Scenario): Promise<Factor[]> {
  if (USE_MOCK) {
    await mockDelay(1500);
    return mockFactors;
  }
  return apiFetch<Factor[]>('/factors/generate', { method: 'POST', body: JSON.stringify(scenario) });
}

export async function getFactor(factorId: string): Promise<Factor> {
  if (USE_MOCK) {
    await mockDelay(200);
    const f = mockFactors.find(f => f.meta.factor_id === factorId);
    if (!f) throw new Error(`因子 ${factorId} 不存在`);
    return f;
  }
  return apiFetch<Factor>(`/factors/${factorId}`);
}

export async function listFactors(status?: string): Promise<Factor[]> {
  if (USE_MOCK) {
    await mockDelay(300);
    if (status) return mockFactors.filter(f => f.status === status);
    return mockFactors;
  }
  const query = status ? `?status=${status}` : '';
  return apiFetch<Factor[]>(`/factors${query}`);
}

File: skill-factor-server/app/api/factor.py
"""因子 CRUD 端点，路径对齐前端 /api/v1/factors。"""

from typing import Optional

from fastapi import APIRouter, HTTPException

from app.models.factor import FactorRecord, FactorStatus
from app.services.factor_pool import get_pool

router = APIRouter(prefix="/factors", tags=["Factors"])


@router.get("")
def list_factors(status: Optional[str] = None):
    """列出因子，可按状态筛选。返回完整 FactorRecord 列表。"""
    pool = get_pool()
    if status:
        try:
            st = FactorStatus(status)
        except ValueError:
            raise HTTPException(400, f"Invalid status: {status}")
        ids = pool.list_status(st)
    else:
        ids = list(pool._by_id.keys())
    return [pool._by_id[fid].model_dump(mode="json") for fid in ids]


@router.get("/{factor_id}")
def get_factor(factor_id: str):
    """获取单个因子完整记录。"""
    record = get_pool().get(factor_id)
    if not record:
        raise HTTPException(404, f"Factor {factor_id} not found")
    return record.model_dump(mode="json")


@router.post("/generate")
def generate_factors(scenario: dict):
    """因子生成（占位，待实现 LLM 调用逻辑）。"""
    raise HTTPException(501, "Factor generation not implemented yet")
